src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,303 @@
+    let pos = 0; // Current global position in the file
+    let lineNum = 1;  // Current line number
+    let linePos = 0; // Current position in the line
+    let res;
+    if (isCode) {
+      pos = this._skipToLicenseHeader(text);
+      pos = this._skipToTextInComments(text, pos);
+      if (pos === null) {
+        return this._createJsonResponse(TOKENS.END, lineNum, linePos, linePos);
+      }
+    }
+    while (pos < text.length) {
+      res = null;
+      for (; pos < text.length; pos++) {
+        const c = text[pos];
+
+        if (c === '\n') {
+          lineNum++;
+          linePos = pos;
+          if (isCode)  {
+            pos = this._skipToTextInComments(text, pos + 1);
+            if (pos === null) {
+              if (res !== null) {
+                yield this._createJsonResponse(res, lineNum, linePos, linePos);
+              }
+              return this._createJsonResponse(TOKENS.END, lineNum, linePos, linePos);
+            }
+          }
+        }
+
+        if (/\s/.test(c)) {
+          if (!res) continue; else break;
+        }
+
+        res = res ? res + c : '' + c;
+      }
+
+      if (res == null) break; // for cases when \n is the last sym
+
+      yield this._createJsonResponse(res, lineNum, linePos, pos);
+    }
+    return this._createJsonResponse(TOKENS.END, lineNum, linePos, pos);
+  }
+
+  /**
+   * Skips all lines that start with '#' and contain only white space characters.
+   * @private
+   * @return {Integer} position
+   */
+  _skipToLicenseHeader(text) {
+    let pos = 0;
+    while (pos < text.length) {
+      const char = text[pos];
+      if (char === '#') {
+        pos = this._skipToNextLine(text, pos);
+        continue;
+      }
+      if (!/\s/.test(char)) {
+        break;
+      }
+      pos++;
+    }
+    return pos;
+  }
+
+  /**
+   * Skips to the next line.
+   * @private
+   * @return {integer} position
+   */
+  _skipToNextLine(text, pos) {
+    while (pos < text.length && text[pos] != '\n') {
+      pos++;
+    }
+    if (pos != text.length) pos++;
+    return pos;
+  }
+
+  _createJsonResponse(token, lineNum, linePos, currentPos) {
+    return {
+      token: token,
+      lineNum: lineNum,
+      linePos: currentPos - linePos
+    };
+  }
+
+  /**
+   * Skips to the text after the single line comments ("//") starting from _pos.
+   *
+   * @private
+   * @return offset from the current position (_pos) to the commented text or null if no comments found.
+   */
+  _skipToTextInComments(text, startPos) {
+    if (startPos + 1 >= text.length) return null;
+    if (text[startPos] != '/' || text[startPos + 1] != '/') {
+      return null;
+    } else {
+      return startPos + 2;
+    }
+  }
+
+  /**
+   * Check path for License mistakes
+   * @param {String} path
+   * @return {[ErrorMessage]}
+   */
+  check(dirpath) {
+    const allFiles = this._getFiles(dirpath, []);
+    const errors = [];
+    errors.push(false); // reserved for LICENSE error
```

Comment:
why not we just push the LICENSE error explicitly here?

Reply:
because we don't want to add it to begining of array to avoid useless unshift, but we want to have License file error first